File: src/main/java/de/noname/pizza4c/datamodel/lieferando/Category.java
package de.noname.pizza4c.datamodel.lieferando;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Category {
    @JsonProperty
    private String id;

    @JsonProperty
    private String name;

    @JsonProperty
    private List<String> description;

    @JsonProperty
    private List<String> productIds;

    @JsonProperty
    public boolean isVegetarian;

    @JsonProperty
    public boolean isVegan;
}

File: frontend/pizza4c-frontend/src/datamodel/restaurant/category.ts
import {selectableVegan} from "../cart/vegan.ts";

export default interface Category {
    id: string;
    name: string;
    description: string[];
    productIds: string[];
    isVegetarian: boolean;
    isVegan: boolean;
}

export function isVeganStateCategory(category: Category, veganState: selectableVegan): boolean {
    switch (veganState){
        case "all":
            return true;
        case "vegetarian":
            return category.isVegetarian;
        case "vegan":
            return category.isVegan;
    }
}